### PR TITLE
standardized BARThez model card

### PR DESCRIPTION
**Pull request by EthanV431**
# What does this PR do?
This standardizes the BARThez model card according to the template outlined in #36979. 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/barthez.md [hunk 1]
```diff
@@ -14,49 +14,93 @@ rendered properly in your Markdown viewer.
...
-">
-</div>
+[BARThez](https://huggingface.co/papers/2010.12321) is the first monolingual French [BART](https://huggingface.co/papers/1910.13461) model, pretrained on a large, adapted French corpus using BART’s denoising objectives. By pretraining both its encoder and decoder, it is uniquely equipped for generative tasks in French, unlike existing French [BERT](https://huggingface.co/papers/1810.04805)‑based models.
```
Comment (stevhliu) [inline docs/source/en/model_doc/barthez.md]:
```suggestion
[BARThez](https://huggingface.co/papers/2010.12321) is a [BART](./bart) model designed for French language tasks. Unlike existing French BERT models, BARThez includes a pretrained encoder-decoder, allowing it to generate text as well. This model is also available as a multilingual variant, mBARThez, by continuing pretraining multilingual BART on a French corpus.
```

File: docs/source/en/model_doc/barthez.md [hunk 2]
```diff
@@ -14,49 +14,93 @@ rendered properly in your Markdown viewer.
...
+
+You can find all of the original BARThez checkpoints under the [BARThez](https://huggingface.co/collections/dascim/barthez-670920b569a07aa53e3b6887) collection.
+
+> [!TIP]
+> This model was contributed by [moussakam](https://huggingface.co/moussakam).
+> Click on the BARThez models in the right sidebar for more examples of how to apply BARThez to different language tasks.
+
+
+The example below demonstrates how to predict the `[MASK]` token with [`Pipeline`], [`AutoModel`], and from the command line.
```
Comment (stevhliu) [inline docs/source/en/model_doc/barthez.md]: It should be `<mask>` token

File: docs/source/en/model_doc/barthez.md [hunk 3]
```diff
@@ -14,49 +14,93 @@ rendered properly in your Markdown viewer.
...
+import torch
+from transformers import pipeline
+
+pipeline = pipeline(
+    task="fill-mask",
+    model="moussaKam/barthez",
+    torch_dtype=torch.float16,
+    device=0
+)
+pipeline("Les plantes produisent [MASK] grâce à un processus appelé photosynthèse.")
```
Comment (stevhliu) [inline docs/source/en/model_doc/barthez.md]:
```suggestion
pipeline("Les plantes produisent <mask> grâce à un processus appelé photosynthèse.")
```

File: docs/source/en/model_doc/barthez.md [hunk 4]
```diff
@@ -14,49 +14,93 @@ rendered properly in your Markdown viewer.
...
+tokenizer = AutoTokenizer.from_pretrained(
+    "moussaKam/barthez",
+)
+model = AutoModelForMaskedLM.from_pretrained(
+    "moussaKam/barthez",
+    torch_dtype=torch.float16,
+    device_map="auto",
+    attn_implementation="sdpa"
+)
+inputs = tokenizer("Les plantes produisent [MASK] grâce à un processus appelé photosynthèse.", return_tensors="pt").to("cuda")
```
Comment (stevhliu) [inline docs/source/en/model_doc/barthez.md]:
```suggestion
inputs = tokenizer("Les plantes produisent <mask> grâce à un processus appelé photosynthèse.", return_tensors="pt").to("cuda")
```

File: docs/source/en/model_doc/barthez.md [hunk 5]
```diff
@@ -14,49 +14,93 @@ rendered properly in your Markdown viewer.
...
-The BARThez model was proposed in [BARThez: a Skilled Pretrained French Sequence-to-Sequence Model](https://huggingface.co/papers/2010.12321) by Moussa Kamal Eddine, Antoine J.-P. Tixier, Michalis Vazirgiannis on 23 Oct,
-2020.
+tokenizer = AutoTokenizer.from_pretrained(
+    "moussaKam/barthez",
+)
+model = AutoModelForMaskedLM.from_pretrained(
+    "moussaKam/barthez",
+    torch_dtype=torch.float16,
+    device_map="auto",
+    attn_implementation="sdpa"
```
Comment (stevhliu) [inline docs/source/en/model_doc/barthez.md]:
```suggestion
```

File: docs/source/en/model_doc/barthez.md [hunk 6]
```diff
@@ -14,49 +14,93 @@ rendered properly in your Markdown viewer.
...
+print(f"The predicted token is: {predicted_token}")
+```
 
-This model was contributed by [moussakam](https://huggingface.co/moussakam). The Authors' code can be found [here](https://github.com/moussaKam/BARThez).
+</hfoption>
+<hfoption id="transformers CLI">
 
-<Tip> 
+```bash
+echo -e "Les plantes produisent [MASK] grâce à un processus appelé photosynthèse." | transformers run --task fill-mask --model moussaKam/barthez --device 0
```
Comment (stevhliu) [inline docs/source/en/model_doc/barthez.md]:
```suggestion
echo -e "Les plantes produisent <mask> grâce à un processus appelé photosynthèse." | transformers run --task fill-mask --model moussaKam/barthez --device 0
```

File: docs/source/en/model_doc/barthez.md [hunk 7]
```diff
@@ -14,49 +14,93 @@ rendered properly in your Markdown viewer.
...
+```
 
-BARThez implementation is the same as BART, except for tokenization. Refer to [BART documentation](bart) for information on 
-configuration classes and their parameters. BARThez-specific tokenizers are documented below.  
+</hfoption>
+</hfoptions>
 
-</Tip>
+## Notes
```
Comment (stevhliu) [inline docs/source/en/model_doc/barthez.md]: No need for this `## Notes` section

File: docs/source/en/model_doc/barthez.md [hunk 8]
```diff
@@ -14,49 +14,93 @@ rendered properly in your Markdown viewer.
...
+- BARThez implementation is the same as BART, except for tokenization. Refer to [BART documentation](bart) for information on 
+configuration classes and their parameters.
 
 ## Resources
```
Comment (stevhliu) [inline docs/source/en/model_doc/barthez.md]: Can remove this `## Resources` section as well

File: docs/source/en/model_doc/barthez.md [hunk 9]
```diff
@@ -14,49 +14,93 @@ rendered properly in your Markdown viewer.
...
-">
-</div>
+[BARThez](https://huggingface.co/papers/2010.12321) is the first monolingual French [BART](https://huggingface.co/papers/1910.13461) model, pretrained on a large, adapted French corpus using BART’s denoising objectives. By pretraining both its encoder and decoder, it is uniquely equipped for generative tasks in French, unlike existing French [BERT](https://huggingface.co/papers/1810.04805)‑based models.
+
+You can find all of the original BARThez checkpoints under the [BARThez](https://huggingface.co/collections/dascim/barthez-670920b569a07aa53e3b6887) collection.
+
+> [!TIP]
+> This model was contributed by [moussakam](https://huggingface.co/moussakam).
+> Click on the BARThez models in the right sidebar for more examples of how to apply BARThez to different language tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/barthez.md]:
```suggestion
> Refer to the [BART](./bart) docs for more usage examples.
```

**Review by stevhliu — APPROVED** [review]
Nice work, thanks!

**Comment by EthanV431** [comment]
Added suggested changes
